packages/p2p/src/components/user/user-rating-recommendation-modal/user-rating-recommendation-modal.jsx

```diff
@@ -0,0 +1,110 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import { Button, Icon, Modal, Text, Rating } from '@deriv/components';
+import { localize } from 'Components/i18next';
+import { user_recommendation as Recommendation } from 'Constants/user-recommendation.js';
+
+const UserRatingRecommendationModal = ({ onClick, is_open, rating, recommendation, user_type }) => {
+    const [user_rating, setUserRating] = React.useState(0);
+    const [user_recommendation, setUserRecommendation] = React.useState(null);
+
+    const handleModalAction = action_status => {
+        if (action_status) {
+            onClick({ rating: user_rating, recommendation: user_recommendation });
+        } else {
+            onClick({ rating, recommendation });
+        }
+    };
+
+    React.useEffect(() => {
+        setUserRating(rating);
+        setUserRecommendation(recommendation);
+    }, [rating, recommendation]);
+
+    return (
+        <Modal
+            has_close_icon={!!user_rating}
+            is_open={is_open}
+            small
+            title={localize('How would you rate this transaction?')}
+            toggleModal={() => handleModalAction(false)}
+        >
+            <Modal.Body>
+                <div className='user-rating-recommendation__layout--vertical'>
+                    <Rating
+                        value={user_rating}
+                        icon_selected={<Icon icon='IcStar' size={16} custom_color='var(--status-warning)' />}
+                        icon_unselected={<Icon icon='IcStar' size={16} custom_color='var(--status-default)' />}
+                        onClick={setUserRating}
+                    />
+                    {user_rating && (
+                        <section className='user-rating-recommendation__layout--vertical'>
+                            <Text>{localize('Would you recommend this {{user_type}}', { user_type })}</Text>
+                            <div className='user-rating-recommendation__layout--horizontal'>
+                                <Button small transparent>
+                                    <div className='user-rating-recommendation__layout--horizontal'>
+                                        <Icon
+                                            icon='IcThumbsUp'
+                                            size={14}
+                                            custom_color={
+                                                user_recommendation === Recommendation.YES
+                                                    ? 'var(--text-prominent)'
+                                                    : 'var(--status-unselect)'
```

Comment:
I recommend using boolean, we don't need YES or NO

Reply:
Will remove the constant as API is returning 1 0r 0